### PR TITLE
Implemented Google Sign In Feature + MVC Structure

### LINKED ISSUE #8: Add Google Sign In with AuthO
Add SSO with Google

### LINKED ISSUE #26: Create MVC Folder Structure
Structure the project using MVC design pattern

### PR DESCRIPTION
**Pull request by chandansgowda**
@jddeep  As per your suggestions, I have implemented Google Sign In without using Webview/Firebase. As of now, I have used my own Google Cloud Console account for development. That has to be changed before deployment. I have also structured the files using MVC design pattern. The UI has to be implemented once the design is confirmed. For now, the required functionality is implemented.

https://user-images.githubusercontent.com/41890434/227315113-d662898f-5134-48e6-ab14-b0f746ded337.mov

Closes #8
Closes #26

### CODE REVIEW

File: lib/services/auth/google_signin.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'dart:io' show Platform;
+import 'package:google_sign_in/google_sign_in.dart';
+
+class GoogleSignInApi {
+  static String _clientId =
+      "375993303437-tsvpf8s7gbs653429utb1p1hgpto4gbv.apps.googleusercontent.com";
```
Comment (jddeep) [inline lib/services/auth/google_signin.dart:6]: Put this in a `constants` file

File: lib/services/auth/google_signin.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  static Future login() async {
+    try {
+      var user = await _googleSignIn.signIn();
+      print(user?.displayName!);
+      return user;
+    } catch (error) {
+      print(error);
+    }
+  }
```
Comment (jddeep) [inline lib/services/auth/google_signin.dart:18]: Do we know how to this user and add it to Auth0 system?

File: lib/services/auth/google_signin.dart (hunk 1 above)
Comment (chandansgowda) [reply lib/services/auth/google_signin.dart]: Done

File: lib/services/auth/google_signin.dart (hunk 2 above)
Comment (chandansgowda) [reply lib/services/auth/google_signin.dart:19]: Yes. Once this and Auth0 PR is merged, we will then create a common User model and initailize that with the loggedInUser info. Also, we'll encode our object into a JSON string and save it using get_storage. That will help us in keeping the user logged in.

Comment (jddeep) [reply lib/services/auth/google_signin.dart:19]: @chandansgowda Are you talking about this PR - https://github.com/AOSSIE-Org/Resonate/pull/15 ?

Comment (chandansgowda) [reply lib/services/auth/google_signin.dart:19]: @jddeep Yes. But that PR will have a lot of conflicts now. I have spoke to him over call. Once my PR gets merged, he will pull it and work on it again. Then once that gets merged, I will work on a common user model.

**Comment by ViAsmit** [comment]
@jddeep can you please review this... so I can start working on next task...

**Review by jddeep — APPROVED** [review]

**Pull request merged**